common/app/controllers/EmailSignupController.scala

```diff
@@ -76,8 +77,17 @@ class EmailSignupController(wsClient: WSClient, val controllerComponents: Contro
     Cached(60)(RevalidatableResult.Ok(views.html.emailLanding(emailLandingPage)))
   }
 
-  def renderForm(emailType: String, listId: Int): Action[AnyContent] = Action { implicit request =>
-    Cached(1.day)(RevalidatableResult.Ok(views.html.emailFragment(emailLandingPage, emailType, listId)))}
+  // This listId can be either an exactTarget listId or the corresponding identity name which is stored in the Identity model
+  def renderForm(emailType: String, listId: String): Action[AnyContent] = Action { implicit request =>
+    Try(listId.toInt) match {
+      case Success(id) => Cached(1.day)(RevalidatableResult.Ok(views.html.emailFragment(emailLandingPage, emailType, id)))
+      case Failure(_) => {
+        val id = EmailNewsletter.fromIdentityName(listId).map(_.listId).getOrElse(-1)
```

Comment:
Might be nicer to explicitly throw/render an error if the list name is invalid.  `BadRequest` or some such.

Reply:
@jfsoul actually do we want to always return some html even if when it submits it will not succeed? So we don't not render anything?

Reply:
I think I would want it to obviously fail when rendered so that whoever is embedding it can see a problem straight away.  What do you think?

Reply:
Yep infact I think your right and it should probably use what we already have and use `WithoutRevalidationResult(NotFound)`